tests/auto/rt_auto.py

```diff
@@ -0,0 +1,510 @@
+    regexhostname : str
+      a regex string to match with login nodes of the machine to determine name
+    workdir : str
+      the directory to perform work in
+    baselinedir : str
+      the directory where baselines are stored on the machine
+    '''
+    def __init__(self, rtdata_obj):
+        self.logger = logging.getLogger("Machine")
+        self.get_machine_info(rtdata_obj)
+        self.machineid = os.environ.get('MACHINE_ID')
+
+    def get_machine_info(self, rtdata_obj):
+        self.logger.info(f'Getting machine information')
+        hostname = socket.gethostname()
+        self.logger.debug(f'hostname is {hostname}')
+        machines = rtdata_obj.get_yaml_subset('machines')
+        regexmachines = machines['regexhostname'].values
+        for i, mach in enumerate(regexmachines):
+            if re.search(mach, hostname):
+                self.logger.info(f'{hostname} is an approved machine')
+                mach_db_info = machines.iloc[i]
+                self.name = mach_db_info['name']
+                self.regexhostname = mach_db_info['regexhostname']
+                self.workdir = mach_db_info['workdir']
+                self.baselinedir = mach_db_info['baselinedir']
+                self.logger.debug(f'self.name: {self.name}\n\
+                    self.regexhostname: {self.regexhostname}\n\
+                    self.workdir: {self.workdir}\n\
+                    self.baselinedir: {self.baselinedir}')
+                self.logger.info(f'Finished getting machine information')
+                return
+            else:
+                continue
+        self.logger.critical(f'Hostname:{hostname} does not match approved list.')
+        raise Exception(f'Hostname:{hostname} does not match approved list.')
+
+class Action:
+    '''
+    This class stores all the information about each action
+    ...
+
+    Attributes
+    ----------
+    name : str
+      the name of the action
+    command : str
+      the command the action will run on the machine
+    callback : str
+      the function to run once the command has completed
+    '''
+    def __init__(self, name, command, callback):
+        self.logger = logging.getLogger("Action")
+        self.name = name
+        self.command = command
+        self.callback = callback
+        self.logger.debug(f'Action object created')
+        self.logger.debug(f'self.name: {self.name}\n\
+            self.command: {self.command}\n\
+            self.callback: {self.callback}')
+
+    def verify_item(self, item, comparable):
+        if item == "name":
+            if re.match(self.name.lower(), comparable.lower()):
+                self.logger.debug(f'Name match found')
+                return True
+        elif item == "command":
+            if re.match(self.name.lower(), comparable.lower()):
+                self.logger.debug(f'Command match found')
+                return True
+        else:
+            self.logger.debug(f'Not a match')
+            return False
```

Comment:
putting it in an `if-else` block doesn't make it any different.
```
    def verify(self, comparable):
           return True if re.match(self.name.lower(), comparable.lower()) else False
```

Reply:
I see my typo: line 138 should show
`if re.match(self.command.lower(), comparable.lower()):`

What about?
```
def verify(self, item, comparable):
    if item == 'name':
        return True if re.match(self.name.lower(), comparable.lower()) else False
    if item == 'command':
        return True if re.match(self.command.lower(), comparable.lower()) else False
```